internal/config/secure/go_keyring.go

```diff
@@ -0,0 +1,231 @@
+// Copyright 2025 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secure
+
+import (
+	"errors"
+	"slices"
+
+	"github.com/zalando/go-keyring"
+)
+
+//go:generate go tool go.uber.org/mock/mockgen -destination=./mocks.go -package=secure github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config/secure KeyringClient
+
+const servicePrefix = "atlascli_"
+
+func createServiceName(profileName string) string {
+	return servicePrefix + profileName
+}
+
+// KeyringClient abstracts keyring operations for easier testing
+type KeyringClient interface {
+	Set(service, user, password string) error
+	Get(service, user string) (string, error)
+	Delete(service, user string) error
+	DeleteAll(service string) error
+}
+
+// DefaultKeyringClient implements KeyringClient using the zalando/go-keyring library
+type DefaultKeyringClient struct{}
+
+func NewDefaultKeyringClient() *DefaultKeyringClient {
+	return &DefaultKeyringClient{}
+}
+
+func (*DefaultKeyringClient) Set(service, user, password string) error {
+	return keyring.Set(service, user, password)
+}
+
+func (*DefaultKeyringClient) Get(service, user string) (string, error) {
+	value, err := keyring.Get(service, user)
+	if err != nil && !errors.Is(err, keyring.ErrNotFound) {
```

Comment:
[q] can we check if keyring is available first on NewStore initialization?

maybe expose a `IsAvailable`